wordpress-mobile/WordPress-Android
# Reader move tab layout to parent part 6 - search

Fixes #11908
Partially fixes #11849

**Merge instructions:**
1. Review this PR
1. ~Make sure https://github.com/wordpress-mobile/WordPress-Android/pull/11899 is merged~
2. ~Change the target branch to master-feature-branch/move-reader-tablayout-to-parent~
3. ~Remove the "Not Ready for merge" label~
4. Merge this PR

This is another PR in a series of PRs which is trying to introduce a regular TabLayout+ViewPager2 in the main Reader view. The current implementation uses FilteredRecyclerView component that has some downsides (eg. swipe gesture doesn't work). These changes will also help us implement the improvements in the ReaderImprovements project.
 
P.S. The known issues/remaining tasks are listed in the ReaderImprovements project board.

Goal of this PR is to re-introduce the search feature in the Reader. I was planning to extract the feature from the ReaderPostListFragment or even rewrite it from scratch. As it turned out this task would be much bigger than I anticipated. The feature is quite complex and contains a lot of workarounds for edge cases. I decided to reuse the existing functionality and just re-wire the navigation.
This solution has one minor downside I'm aware of -> when the user clicks on the search icon a new screen is displayed and the default SearchView animations isn't played. This could be potentially solved by using `fragment transition animations` but I decided against it to keep the code as simple as possible. The fragment/activity transition looks good enough imo. 

To test:
1. Open Reader
2. Click on Search
3. Type something
4. Click on the clear button
5. Type something
6. Click on Done
7. Notice the search is performed
8. Scroll and switch tabs
9. Rotate your device and make sure the scroll position is retained and the app doesn't crash :P
10. Restart the app
11. Open Reader
12. Click on Search
13. Notice the previously searched query is being suggested
14. Type a first letter of the same query
15. Notice auto-completion works

Continue testing whatever you can think of.

PR submission checklist:
